odbc/src/api/data_tests.rs

```diff
@@ -0,0 +1,1012 @@
+                str_val_test(14, "{\"$oid\":\"63448dfed38427a35d534e40\"}");
+                str_val_test(
+                    15,
+                    "{\"$regularExpression\":{\"pattern\":\"hello .* world\",\"options\":\"\"}}",
+                );
+                str_val_test(16, "hello world!");
+            }
+
+            {
+                let mut null_val_test = |col: u16| {
+                    assert_eq!(
+                        SqlReturn::SUCCESS,
+                        SQLGetData(
+                            stmt_handle as *mut _,
+                            col,
+                            CDataType::Char,
+                            char_buffer,
+                            buffer_length,
+                            out_len_or_ind,
+                        )
+                    );
+                    assert_eq!(odbc_sys::NULL_DATA, *out_len_or_ind);
+                };
+
+                null_val_test(13);
+                null_val_test(17);
+            }
+            let _ = Box::from_raw(char_buffer);
+        }
+    }
+
+    #[test]
+    fn sql_get_bool_data() {
+        use crate::api::functions::SQLGetData;
+        use odbc_sys::CDataType;
+        let mut stmt = Statement::with_state(std::ptr::null_mut(), StatementState::Allocated);
+        stmt.mongo_statement = Some(Box::new((*MQ).clone()));
+        let stmt_handle: *mut _ = &mut MongoHandle::Statement(RwLock::new(stmt));
+        unsafe {
+            assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt_handle as *mut _,));
+            let buffer: *mut std::ffi::c_void = Box::into_raw(Box::new([0u8; 10])) as *mut _;
+            let buffer_length: isize = 10;
+            let out_len_or_ind = &mut 0;
+            {
+                let mut bool_val_test = |col: u16, expected: bool| {
+                    assert_eq!(
+                        SqlReturn::SUCCESS,
+                        SQLGetData(
+                            stmt_handle as *mut _,
+                            col,
+                            CDataType::Bit,
+                            buffer,
+                            buffer_length,
+                            out_len_or_ind,
+                        )
+                    );
+                    assert_eq!(1, *out_len_or_ind);
+                    assert_eq!(expected, *(buffer as *const bool));
+                };
+
+                bool_val_test(1, false);
+                bool_val_test(2, false);
+                bool_val_test(3, true);
+                bool_val_test(4, true);
+                bool_val_test(5, false);
+                bool_val_test(6, true);
+                bool_val_test(7, true);
+                bool_val_test(8, true);
+                bool_val_test(9, false);
+                bool_val_test(10, false);
+                bool_val_test(11, false);
+                bool_val_test(12, false);
+                bool_val_test(14, false);
+                bool_val_test(15, false);
+                bool_val_test(16, true);
```

Comment:
```suggestion
                bool_val_test(1, false); // array
                bool_val_test(2, false); // binary
                bool_val_test(3, true); // bool
                bool_val_test(4, true); // datetime
                bool_val_test(5, false); // doc
                bool_val_test(6, true); // double
                bool_val_test(7, true); // int32
                bool_val_test(8, true); // int64
                bool_val_test(9, false); // js
                bool_val_test(10, false); // js_w_s
                bool_val_test(11, false); // max_key
                bool_val_test(12, false); // min_key
                bool_val_test(14, false); // oid
                bool_val_test(15, false); // regex
                bool_val_test(16, true); // string
```
This one is really up to you. Ideally we'd have this comment for each test below (long, int, double, etc.) but I'm not sure if it's totally worth it 🤷‍♀️ Maybe I'm just finding it useful while reviewing because it's helping me ensure there are no silly typo-style bugs. It could still be nice for future readers to have as much information in view _in_ the tests. Your call on this and the other tests with the same set-up.

Reply:
Oh yea, that's a good idea too! Probably even better tbh